src/super.c

```diff
@@ -344,11 +354,32 @@ int scoutfs_read_super(struct super_block *sb,
 		goto out;
 	}
 
-	*super_res = *super;
 	ret = 0;
 out:
-	kfree(super);
-	return ret;
+	if (ret < 0) {
+		kfree(super);
+		return ERR_PTR(ret);
```

Comment:
Just set super to ERR_PTR(ret) and let it fall through to returning it rather than having branches and unbalanced braces.